pages/index.js

```diff
@@ -27,8 +27,8 @@ import NewsroomSection from '../components/newsroom/NewsroomSection'
 import { languageDetection } from "../lib/i18n";
 
 function HomePage() {
-  //To be enabled in a future PR
-  //languageDetection();
+  
+  languageDetection();
```

Comment:
I think this function should be inside `useEffect` because the initial rendering of the Homepage takes place (in "EN") first before changing the language of the page.

Reply:
Actually `languageDetection()` is itself a hook so cannot be called inside `useEffect()`

Reply:
Add a preloader same as tools dashboard page.

Reply:
Done 🚀